hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/state/AssignmentUIState.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.management.state;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.hawkbit.ui.common.entity.DistributionSetIdName;
+import org.eclipse.hawkbit.ui.common.entity.TargetIdName;
+
+/**
+ * State keeping track of the distribution set assignments that were done in the
+ * UI.
+ */
+public class AssignmentUIState implements Serializable, Iterable<DistributionSetIdName> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final Map<DistributionSetIdName, Set<TargetIdName>> assignments = new HashMap<>();
```

Comment:
double check if Map<TargetIdName, Set<DistributionSetIdName>> would not be better

Reply:
Removed AssignmentUIState completely